LayoutTests/http/wpt/webcodecs/audioData-copyTo.html

```diff
@@ -0,0 +1,79 @@
+<!DOCTYPE html>
+<html>
+<header>
+<script src='/resources/testharness.js'></script>
+<script src='/resources/testharnessreport.js'></script>
+</header>
+<body>
+<script>
+test(t => {
+  const data = new AudioData({
+    format: 'f32-planar',
+    sampleRate: 48000,
+    numberOfChannels: 1,
+    numberOfFrames: 5,
+    data: new Float32Array([1, 2, 3, 4, 5]),
+    timestamp: 0,
+  });
+  const output = new Float32Array(data.numberOfFrames);
+  data.copyTo(output.subarray(1, 4), {
+    planeIndex: 0,
+    frameOffset: 1,
+    frameCount: 3,
+  });
+  data.close();
+  assert_array_equals(output, [0, 2, 3, 4, 0], 'only 3 middle elements are copied');
+}, 'Test that memcpy branch of AudioData.copyTo only copies frameCount amount of mono frames from frameOffset');
+
+test(t => {
+  const data = new AudioData({
+    format: 'f32-planar',
+    sampleRate: 48000,
+    numberOfChannels: 2,
+    numberOfFrames: 5,
+    data: new Float32Array([
+      1, 2, 3, 4, 5, // left channel
+      6, 7, 8, 9, 10, // right channel
+    ]),
+    timestamp: 0,
+  });
+  const output = new Float32Array(data.numberOfFrames * 2);
+  data.copyTo(output.subarray(1, 4), {
+    planeIndex: 0,
+    frameOffset: 1,
+    frameCount: 3,
+  });
+  data.copyTo(output.subarray(data.numberOfFrames + 1, data.numberOfFrames + 4), {
+    planeIndex: 1,
+    frameOffset: 1,
+    frameCount: 3,
+  });
+  data.close();
+  assert_array_equals(output, [
+    0, 2, 3, 4, 0, // left channel
+    0, 7, 8, 9, 0, // right channel
+  ], 'only 3 middle elements are copied in both channels');
+}, 'Test that memcpy branch of AudioData.copyTo only copies frameCount amount of stereo frames from frameOffset');
+
+test(t => {
+  const data = new AudioData({
+    format: 'f32',
+    sampleRate: 48000,
+    numberOfChannels: 1,
+    numberOfFrames: 5,
+    data: new Float32Array([1, 2, 3, 4, 5]),
+    timestamp: 0,
+  });
+  const output = new Float32Array(data.numberOfFrames);
+  data.copyTo(output.subarray(1, 4), {
+    planeIndex: 0,
+    frameOffset: 1,
+    frameCount: 3,
+    format: 'f32',
+  });
+  data.close();
+  assert_array_equals(output, [0, 2, 3, 4, 0], 'only 3 middle elements are copied');
+}, 'Test that memcpy branch of AudioData.copyTo only copies frameCount amount of interleaved mono frames from frameOffset');
```

Comment:
In all honesty it was a bit of an accidental discovery that we can test that memcpy is used because the unoptimised path is clamping the values between -1 and 1

https://github.com/WebKit/WebKit/blob/a6f68ea69e73ae1619a17c2a6f4abc4eb8de8105/Source/WebCore/platform/AudioSampleFormat.h#L150

If that's too implementation-specific, I'm happy to remove this test and change all values to be in -1...1 range.